src/PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/Ricci.cpp

```diff
@@ -0,0 +1,167 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/Ricci.hpp"
+
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tags/TempTensor.hpp"
+#include "DataStructures/TempBuffer.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+/// \cond
+namespace GeneralizedHarmonic {
+template <size_t VolumeDim, typename Frame, typename DataType>
+void spatial_ricci_tensor(
+    const gsl::not_null<tnsr::ii<DataType, VolumeDim, Frame>*> ricci,
+    const tnsr::iaa<DataType, VolumeDim, Frame>& phi,
+    const tnsr::ijaa<DataType, VolumeDim, Frame>& deriv_phi,
+    const tnsr::II<DataType, VolumeDim, Frame>&
+        inverse_spatial_metric) noexcept {
+  destructive_resize_components(ricci, get_size(get<0, 0, 0>(phi)));
+
+  TempBuffer<tmpl::list<::Tags::TempIjj<0, VolumeDim, Frame, DataType>,
+                        ::Tags::TempijK<0, VolumeDim, Frame, DataType>,
+                        ::Tags::Tempi<0, VolumeDim, Frame, DataType>,
+                        ::Tags::Tempi<1, VolumeDim, Frame, DataType>,
+                        ::Tags::TempI<0, VolumeDim, Frame, DataType>,
+                        ::Tags::TempI<1, VolumeDim, Frame, DataType>>>
+      local_buffer(get_size(get<0, 0>(inverse_spatial_metric)));
+
+  // New variable to avoid recomputing in nested loops
+  auto& spatial_phi_Ijj =
```

Comment:
please name this `half_spatial_phi_Ijj` because of the half

Reply:
renamed